src/provider/systemdmanager/systemdunit.rs

```diff
@@ -64,39 +127,79 @@ impl SystemDUnit {
 
         unit.name = format!("{}{}", name_prefix, trimmed_name);
 
-        unit.add_property(SECTION_UNIT, "Description", &unit.name.clone());
+        unit.add_property(Section::Unit, "Description", &unit.name.clone());
 
         unit.add_property(
-            SECTION_SERVICE,
+            Section::Service,
             "ExecStart",
-            &SystemDUnit::get_command(container, pod_state)?,
+            &SystemDUnit::get_command(container, template_data, package_root)?,
         );
 
-        let env_vars = SystemDUnit::get_environment(container, pod_state)?;
-
-        for (name, value) in env_vars {
-            unit.add_env_var(&name, &value);
+        let env_vars = SystemDUnit::get_environment(container, service_name, template_data)?;
+        if !env_vars.is_empty() {
+            let mut assignments = env_vars
+                .iter()
+                .map(|(k, v)| format!("\"{}={}\"", k, v))
```

Comment:
Did you choose the format `Environment="<varname>=<varcontent" "<varname2>=<varcontent2>" ... ` for a specific reason?
I usually prefer spreading multiple vars over multiple lines `Environment="<varname>=<varcontent"\nEnvironment="<varname2>=<varcontent2>"` out of fear of a single line growing too long.

Reply:
Sounds good to me.

Reply:
I added a TODO.